### PR TITLE
ICMSLST-1086 Add search logic for several fields

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

File: web/domains/case/_import/models.py [hunk 1]
```diff
@@ -158,7 +158,9 @@ class Meta:
                 name="IA_search_case_reference_idx",
                 opclasses=["text_pattern_ops"],
             ),
-            models.Index(fields=["-submit_datetime"], name="IA_submit_datetime_idx"),
+            models.Index(
+                *[models.Q(submit_datetime__isnull=False)], name="IA_submit_datetime_notnull_idx"
```
Comment (mikeylockwood) [inline web/domains/case/_import/models.py:162]: Can this first arg just be `models.Q(submit_datetime__isnull=False)`

File: web/migrations/0001_initial.py [hunk 2]
```diff
@@ -6139,7 +6139,9 @@ class Migration(migrations.Migration):
         ),
         migrations.AddIndex(
             model_name="importapplication",
-            index=models.Index(fields=["-submit_datetime"], name="IA_submit_datetime_idx"),
+            index=models.Index(
+                models.Q(("submit_datetime__isnull", False)), name="IA_submit_datetime_notnull_idx"
```
Comment (mikeylockwood) [inline web/migrations/0001_initial.py:6143]: Double parenthesis on the `Q` call

**Review by mikeylockwood — APPROVED** [review]

File: web/migrations/0001_initial.py (hunk 2 above)
Comment (mikeylockwood) [reply web/migrations/0001_initial.py:6143]: Obviously the first arg is a tuple

Comment (MattHolmes123) [reply web/migrations/0001_initial.py:6143]: Django is doing this not me :)

File: web/domains/case/_import/models.py (hunk 1 above)
Comment (MattHolmes123) [reply web/domains/case/_import/models.py:162]: Yer it totally can, I will change it tomorrow thanks

**Pull request merged**